### PR TITLE
LPS25H pressure sensor driver

### PR DESCRIPTION
**Pull request by ReneHerthel**
- [Reference](http://www.st.com/st-web-ui/static/active/en/resource/technical/document/datasheet/DM00066332.pdf) for LPS25H.
- Basic driver functionality
  - Initialize device
  - Read pressure output
  - Read temperature output
  - Enable / Disable device
- Original [issue](https://github.com/RIOT-OS/RIOT/issues/3020) of limifrog-v1 port

### CODE REVIEW

**Comment by OlegHahm** [comment]
Apparently your git client isn't configured using one of your Github mail addresses.

**Comment by ReneHerthel** [comment]
Rebase

**Comment by OlegHahm** [comment]
@PeterKietzmann, what's the state?

**Comment by PeterKietzmann** [comment]
@katezilla please review and test this one as well

**Comment by katezilla** [comment]
@ReneHerthel can you rebase please

**Comment by PeterKietzmann** [comment]
@katezilla please don't comment in separate commits

**Comment by katezilla** [comment]
oh, sorry. was shown to me as one.

**Comment by ReneHerthel** [comment]
rebased

File: drivers/include/lps25h.h [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    LPS25H_ODR_25HZ  = 0x40       /**< sample with 25Hz */
+} lps25h_odr_t;
+
+/**
+ * @brief Initialize a given LPS25H pressure sensor
+ *
+ * @param[out]  dev     device descriptor of the sensor
+ * @param[in]   i2c     I2C bus the sensor is connected to
+ * @param[in]   address the sensor's address on the I2C bus
+ * @param[in]   rate    internal sampling rate of the sensor
```
Comment (OlegHahm) [inline drivers/include/lps25h.h]: `s/rate/odr`

Comment (ReneHerthel) [inline drivers/include/lps25h.h]: - [ ]

**Comment by ReneHerthel** [comment]
rebase

**Comment by OlegHahm** [comment]
What the status here? Any chance to get this merged until next Wednesday?

**Comment by katezilla** [comment]
tested changes. Now it doesn't crash anymore... but values are not right.
temperature shows 42°C and pressure value is alway 0, no matter what i try.

@OlegHahm realively low, as the values are completly wrong.

**Comment by OlegHahm** [comment]
Thanks.

**Comment by kYc0o** [comment]
Feature freeze -> postponed.

**Comment by PeterKietzmann** [comment]
BTW @katezilla, @ReneHerthel what took you so long? IIRC this PR once was short before merging. How about you get this up to date at tomorrows Hack'n'ACK :-)?

**Comment by aabadie** [comment]
@kYc0o, we can test this sensor as it present on the [STM MEMS extension board](http://www.st.com/content/st_com/en/products/ecosystems/stm32-open-development-environment/stm32-nucleo-expansion-boards/stm32-ode-sense-hw/x-nucleo-iks01a1.html) we have.

**Comment by ReneHerthel** [comment]
Rebase.

File: drivers/lps25h/include/lps25h-internal.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+/** RESERVED 0x11-0x1F                */
+#define LPS25H_REG_CTRL_REG1            (0x20)
+#define LPS25H_REG_CTRL_REG2            (0x21)
+#define LPS25H_REG_CTRL_REG3            (0x22)
+#define LPS25H_REG_CTRL_REG4            (0x23)
+#define LPS25H_REG_INT_CFG              (0x24)
+#define LPS25H_REG_INT_SOURCE           (0x25)
+/** RESERVED 0x26                     */
+#define LPS25H_REG_STATUS_REG           (0x27)
+#define LPS25H_REG_PRESS_POUT_XL        (0x28)
```
Comment (katezilla) [inline drivers/lps25h/include/lps25h-internal.h]: dont you mean LPS25H_REG_PRESS_OUT_XL

File: drivers/lps25h/lps25h.c [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+int lps25h_read_pres(lps25h_t *dev)
+{
+    char tmp[3] = {0, 0, 0};
+    int16_t val = 0;
+
+    i2c_acquire(dev->i2c);
+    i2c_read_regs(dev->i2c, dev->addr, LPS25H_REG_PRESS_POUT_XL, &tmp[0], 3);
+    i2c_release(dev->i2c);
+
+    val = (tmp[2] << 16) | (tmp[1] << 8) | tmp[0];
```
Comment (katezilla) [inline drivers/lps25h/lps25h.c]: are you sure about this 24bit value in a 16bit variable?

**Comment by katezilla** [comment]
trying to figure out why the temperature value keeps counting  up

File: drivers/lps25h/lps25h.c [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    int16_t val = 0;
+    float res = 0.0;
+
+    i2c_acquire(dev->i2c);
+    i2c_read_regs(dev->i2c, dev->addr, LPS25H_REG_TEMP_OUT_L, &tmp[0], 2);
+    i2c_release(dev->i2c);
+
+    val = (tmp[1] << 8) | tmp[0];
+
+        DEBUG("LPS25H: raw temperature: %i\n", val);
```
Comment (katezilla) [inline drivers/lps25h/lps25h.c]: please fit intent here

File: drivers/lps25h/lps25h.c [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  * @param[in] value    value which represents a two's complement number
+  * @param[in] mask     bitmask to choose between the single bits
+  *
+  * @return             the converted signed number of 'value'
+  */
+static inline int _twos_complement(int value, int mask)
+{
+    if (value & mask) {
+        value = ~(value & !(mask)) + 1;
+        return ~(value & !(mask));
```
Comment (katezilla) [inline drivers/lps25h/lps25h.c]: i suggest using ~ instead of ! as you want to toggle the bits in mask

**Comment by ReneHerthel** [comment]
Rebase.

**Comment by ReneHerthel** [comment]
Hi @PeterKietzmann and @katezilla,

I've done here some fixes to. Now I'm waiting for Murdock to lock if there are some building errors. But this PR seems to be blocked? I'm not sure how to start it.

**Comment by PeterKietzmann** [comment]
@ReneHerthel I assume you will have the same problem here as in #3263 but I will add the CI label anyway so you can have a look at the build output.

**Comment by PeterKietzmann** [comment]
An one other thing: Did you encounter the same problems that Katja described and did you fix them???

**Comment by ReneHerthel** [comment]
@PeterKietzmann I fixed all things she said except the one thing, where she said "trying to figure out why the temperature value keeps counting up".
The sensors temperature output in the console behaves like a counter in two directions - increment and decrement.
Sometimes, it goes up to ~80C° and then it goes back to ~20°. But only, if I hold it in my hands.
I have to check if the sensor is that sensitive, or I just build unconsciously a counter in one of the functions.

**Comment by katezilla** [comment]
@ReneHerthel i had that 'counter-like' bug in our office with normal temperatures. It counted  also with an overflow but maybe it changed after your fixes.

**Comment by aabadie** [comment]
ping @ReneHerthel, any update on this one ? it seems to be nearly ready.

**Comment by aabadie** [comment]
ping !

**Comment by aabadie** [comment]
Closing in favor of #7267. Hope this is fine.

**Pull request closed without merging**